src/controllers/controllerengine.cpp

```diff
@@ -784,14 +784,15 @@ void ControllerEngine::trigger(QString group, QString name) {
    -------- ------------------------------------------------------ */
 QScriptValue ControllerEngine::connectControl(QString group, QString name,
                                               QScriptValue callback, bool disconnect) {
-    ConfigKey key(group, name);
     ControlObjectThread* cot = getControlObjectThread(group, name);
-    QScriptValue function;
+    ControllerEngineConnection conn;
+    conn.key = ConfigKey(group, name);
+    conn.ce = this;
```

Comment:
can you rename this to something more significant like "conn.parent = this" or similar.